reference/enterprise-api-ref/cmdb-api.markdown

```diff
@@ -0,0 +1,317 @@
+    }
+}
+```
+
+## Create new configuration
+
+**URI:** https://hub.cfengine.com/cmdb
+
+**Method:** POST
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+* **classes** *(JSON object)*
+  The format is a JSON object where the key is class name and value is another JSON object
+  with an optional `comment` property.
+  Example:
+```
+{
+   "classes":{
+      "My_class":{
+         
+      },
+      "My_class2":{
+         "comment":"comment body"
+      }
+   }
+}
+```
+  
+* **variables** *(JSON object)*
+  The format is a JSON object where the key is variable name and value is another JSON object
+  with a required `value` property and an optional `comment` property.
+  Example:
+```
+{
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
+      },
+      "HubCMDB:My.hostname":{
+         "value":"host1.cfengine.com",
+         "comment":"My hostname should be set to this"
+      }
+   }
+}
+```
+
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/cmdb \
+  -H 'content-type: application/json' \
+  -d '{
+   "hostkey":"SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab",
+   "classes":{
+      "My_class2":{
+         "comment" : ""
+      },
+      "My_class": {}
+   },
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
+      },
+      "HubCMDB:My.hostname":{
+         "value":"host1.cfengine.com",
+         "comment":"My hostname should be set to this"
+      }
+   }
+}'
+```
+
+**Example response:**
+
+```
+HTTP 201 Created
+```
+
+## Update configuration
+
+**URI:** https://hub.cfengine.com/cmdb/:hostkey
+
+**Method:** PATCH
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+* **classes** *(JSON object)*
+  The format is a JSON object where the key is class name and value is another JSON object
+  with an optional `comment` property.
+  Example:
+```
+{
+   "classes":{
+      "My_class":{
+         
+      },
+      "My_class2":{
+         "comment":"comment body"
+      }
+   }
+}
+```
+
+If you need to delete all classes from host you need to set null value:
+```
+{
+    "classes": null
+}
+```
+If your request body misses classes then the previous value will be preserved.
+
+* **variables** *(JSON object)*
+  The format is a JSON object where the key is variable name and value is another JSON object
+  with a required `value` property and an optional `comment` property.
+  Example:
+```
+{
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
+      },
+      "HubCMDB:My.hostname":{
+         "value":"host1.cfengine.com",
+         "comment":"My hostname should be set to this"
+      }
+   }
+}
+```
+If you need to delete all variables from host you need to set null value:
+```
+{
+    "variables": null
+}
+```
+If your request body misses variables then the previous value will be preserved.
```

Comment:
This makes me wonder, do I have to provide _all_ previously defined variables when setting a new one?

Reply:
@aleksandrychev ?

Reply:
in this case, yes, but since I opened this PR I've added new API calls to add/update/delete specific items. I will add them later this week.